source/adapters/level_zero/context.hpp

```diff
@@ -333,10 +334,38 @@ struct ur_context_handle_t_ : _ur_object {
         return WithProfiling ? &EventCaches[2] : &EventCaches[3];
       }
     }
+  };
+  auto getCounterBasedEventCache(bool UsingImmediateCmdList,
+                                 ur_device_handle_t Device) {
+    if (UsingImmediateCmdList) {
+      if (Device) {
+        auto EventCachesMap = &EventCachesDeviceMap[4];
+        if (EventCachesMap->find(Device) == EventCachesMap->end()) {
```

Comment:
can you please explain how this structure works ?
why we have some sizes fixed like "4", while you also add code that resizes the map ?

Reply:
EventCachesDeviceMap is fixed, however, the EventCacheMap that is contained within EventCachesDeviceMap is variable and can be used to keep track of the cache of that type and can also change in size.

Reply:
This is not maintainable, we cannot have magic numbers all around the place with no explanation whatsoever.